balancer/weightedtarget/weightedtarget_test.go

```diff
@@ -569,7 +575,9 @@ func (s) TestWeightedTarget_TwoSubBalancers_MoreBackends(t *testing.T) {
 	}
 
 	// Turn sc1's connection down.
-	wtb.UpdateSubConnState(sc1, balancer.SubConnState{ConnectivityState: connectivity.TransientFailure})
+	wtb.UpdateSubConnState(sc1, balancer.SubConnState{
```

Comment:
Instead of declaring the errors as globals, prefer declaring them here. And use error values directly instead of strings.

```golang
errSC1 := errors.New("error from sc1")
wtb.UpdateSubConnState(sc1, balancer.SubConnState{
	ConnectivityState: connectivity.TransientFailure,
	ConnectionError:  errSC1,
})
```

Similarly define an errSC4 for sc4. And when doing the pick, you could do:
```golang
for i := 0; i < 5; i++ {
	_, err := p.Pick(balancer.PickInfo{}); 
	if err == nil || (!strings.Contains(err.Error(), errSC1.Error()) &&  !strings.Contains(err.Error(), errSC4.Error()) {
		t.Fatalf("Pick() returned error: %v, want error: %v or %v",err, errSC1, errSC4)
	}
}
```

Note that in the above error message passed to `t.Fatalf()`, we specify got before want. See: go/go-style/decisions#got-before-want

Also, with this approach, we are concerned about the `last connection error`. That is actually coming from the base balancer and we dont want to rely on that here. As long as the returned error contains the error message that our picker returned, we should be good.

Reply:
Also, please follow the same pattern for the other tests that you have modified as well.

Reply:
We generally prefer `errors.New()` when the error message does not contain any formatting directive.

Reply:
makes sense. updating this in the revision